compatibility/sandbox-migration/runner/Migration/KeyTransfer.hs

```diff
@@ -0,0 +1,99 @@
+-- Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+module Migration.KeyTransfer (test) where
+
+import qualified Data.Aeson as A
+import qualified Data.Text as T
+import GHC.Generics (Generic)
+import System.IO.Extra
+import System.Process
+
+import Migration.Types
+
+test :: FilePath -> FilePath -> Test ([Tuple2 (ContractId Asset) Asset], [Transaction]) Result
+test step modelDar = Test {..}
+  where
+    initialState = ([], [])
+    executeStep sdkVersion host port _state = withTempFile $ \outputFile -> do
+        let suffix = getSdkVersion sdkVersion
+        callProcess step
+            [ "--host=" <> host
+            , "--port=" <> show port
+            , "--output", outputFile
+            , "--dar=" <> modelDar
+            , "--test=key-transfer," <> T.unpack (getParty testOwner) <> "," <> T.unpack (getParty testReceiver) <> "," <> suffix
+            ]
+        either fail pure =<< A.eitherDecodeFileStrict' outputFile
+    validateStep _sdkVersion prev Result{..} =
+        Right prev
+
+    testOwner :: Party
+    testOwner = Party "owner"
+    testReceiver :: Party
+    testReceiver = Party "receiver"
+
+-- The datatypes are defined such that the autoderived Aeson instances
+-- match the DAML-LF JSON encoding.
+
+newtype ContractId t = ContractId T.Text
```

Comment:
There are a couple of duplicates types with the other test (at least `ContractId`, `Party` and `Tuple2`). Maybe worth moving them to a separate module?

Reply:
Yeah, I was hoping to make it all generic, including the `Event`, but as you confirmed, it's not that easy.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/6298/commits/3e3105e643882338319360bb03a2242bf768a0d9